File: lib/Emailer.php
<?php
class EmailTemplate
{
    const TEMPLATES_FOLDER = '/system/email_templates';
    const EMAILS_FOLDER = '/local/logs/emails';

    var $variables = array();
    var $path_to_file= array();
    var $template_name = '';

    function __construct($template_name)
    {
        $this->template_name = $template_name;
    }

    public function __set($key,$val)
    {
        $this->variables[$key] = $val;
    }


    private function _compile($file_name_postfix)
    {
        global $app;
        $path_to_file = $app->home_path().EmailTemplate::TEMPLATES_FOLDER.'/'.$this->template_name.$file_name_postfix.'.'.$app->lang;

         if(!file_exists($path_to_file))
         {
             log_error('EmailTemplate: Template File not found: '.$path_to_file);
             return false;
         }

        ob_start();

        extract($this->variables);
        include $path_to_file;

        $content = ob_get_contents();
        ob_end_clean();

        return $content;
    }

    function compile_body()
    {
        return $this->_compile('_body');
    }

     function compile_subject()
    {
        return $this->_compile('_subject');
    }

}

class Emailer
{
    var $recipients = array();
    var $template;
    var $template_subject;

    public function __construct($to = false)
    {
        if($to !== false)
        {
            if(is_array($to))
            {
                foreach($to as $_to){ $this->recipients[$_to] = $_to; }
            }else
            {
                $this->recipients[$to] = $to; //1 Recip
            }
        }
    }

    function set_template( $template)
    {

        $this->template = $template;
    }

    function send() 
    {        
        global $app;

        $subject = $this->template->compile_subject();
        $body = $this->template->compile_body();

        // SEND REAL EMAIL USING GMAIL
        $gmail = $app->config->gmail;
        if($gmail->enabled){
            ///
            require_once '../lib/vendor/swiftmailer/swiftmailer/lib/swift_required.php';

            $transport = Swift_SmtpTransport::newInstance('smtp.gmail.com', 465, "ssl")
              ->setUsername($gmail->auth_name)
              ->setPassword($gmail->auth_key);

            $mailer = Swift_Mailer::newInstance($transport);

            $message = Swift_Message::newInstance($subject)
              ->setFrom(array('mbazarov@gmail.com' => 'Mailer'))
              ->setTo($this->recipients)
              ->setBody( $body );

            $result = $mailer->send($message);
            if($result===false)
            {
                log_error('Emailer: send : failed');
                return false;
            }
            log_ok('Emailer: send : sent ok');
    
        }

        // DUMP EMAIL INFO FILE
        {

        }

        
        return true;

    }
}
?>

File: lib/Teplo500/Emailer.py
import os.path
from mako.template import Template
from Teplo500.utils import *

TEMPLATES_FOLDER = '/system/email_templates'
EMAILS_FOLDER = '/local/logs/emails'


class EmailTemplate:
    
    def __init__(self, template_name):
        self.variables = {}
        self.path_to_file= ''
        self.template_name = template_name


    def __set(self, key, val):
        self.variables[key] = val
    

    def _compile(self, file_name_postfix):
        path_to_file = app.home_path() + TEMPLATES_FOLDER + '/' + self.template_name + file_name_postfix+'.'+app.lang

        mytemplate = Template(filename=path_to_file)
        if not mytemplate:        
            log_error('AbstractPage: Template File not found: '+path_to_file)
            return None   
 
        return mytemplate.mytemplate.render_unicode(self.variables)
    
    def compile_body(self):
        return self._compile('_body')
    

    def compile_subject(self):
        return this._compile('_subject')

class Emailer:

    def __construct(self,to):    
        self.template = None
        self.template_subject = None

        self.recipients = {
            to:to ##1 Recip        
        }
    

    def set_template(self, template):
        self.template = template
    

    def send(self):     
        subject = self.template.compile_subject()
        body = self.template.compile_body()

        ## SEND REAL EMAIL USING GMAIL
        gmail = app.config['gmail']
        if gmail['enabled']:
            ## TODO: link to some emailer
            ## gmail['auth_name']            
            ## gmail['auth_key']
            ## subject
            ## self.recipients
            ## body 

            log_ok('Emailer: send : sent ok');
        
        return True
